### PR TITLE
Add functional raw device events test

### PR DESCRIPTION
**Pull request by mmkayPL**

### CODE REVIEW

**Comment by tomaszrybicki** [comment]
retest this please

**Comment by tomaszrybicki** [comment]
add to whitelist

File: tests/functional/trace_io_events.py [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+            events = iotrace.get_trace_events(trace_path)
+            events_parsed = iotrace.parse_json(events)
+            result = any(
+                'io' in event and
+                'operation' in event['io'] and
+                event['io']['operation'] == 'Write' and
+                # LBA 0 events don't have a lba field, so skip them
+                'lba' in event['io'] and
+                int(event['io']['lba']) == int(2 * write_length.get_value() / 512) and
+                int(event['io']['len']) == int(write_length.get_value() / 512) and
```
Comment (tomaszrybicki) [inline tests/functional/trace_io_events.py]: Please use a  variable for the "2" here - I assume it is the tested LBA?

File: tests/functional/trace_io_events.py [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    end_lba = start_lba + number_buckets * bucket_size
+    for disk in TestRun.dut.disks:
+        io_len = Size(1, disk.block_size)
+        with TestRun.step("Start tracing"):
+            iotrace.start_tracing([disk.system_path])
+            time.sleep(5)
+        with TestRun.step("Send write commands"):
+            for bucket_index in range(number_buckets):
+                for i in range(bucket_index + 1):
+                    seek = (start_lba + bucket_index * bucket_size + randrange(bucket_size)) * 512 / disk.block_size.get_value()
```
Comment (tomaszrybicki) [inline tests/functional/trace_io_events.py]: Please extract 512 to e.g. variable or helper function with comment on why is it needed (iotrace to different block size conversion)

File: tests/functional/trace_io_events.py [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+                'operation' in event['io'] and
+                event['io']['operation'] == 'Write' and
+                # LBA 0 events don't have a lba field, so skip them
+                'lba' in event['io'] and
+                int(event['io']['lba']) == int(2 * write_length.get_value() / 512) and
+                int(event['io']['len']) == int(write_length.get_value() / 512) and
+                f"/dev/{event['device']['name']}" == disk.system_path for
+                event in events_parsed)
+            if not result:
+                raise Exception("Could not find write event")
```
Comment (Ostrokrzew) [inline tests/functional/trace_io_events.py]: When TF update would be merged, I think You should consider usage of TestRun.LOGGER.error() or TestRun.fail() if test should be stopped right in this moment, when fails.
Watch out! TestRun.LOGGER.error() returns PASSED result, but this should be changed in near future.
This is for all 'raise Exception' lines in test steps.

**Pull request merged**